src/libsyncengine/propagation/executor/executorworker.cpp

```diff
@@ -1559,6 +1561,8 @@ bool ExecutorWorker::enoughLocalSpace(SyncOpPtr syncOp) {
 }
 
 void ExecutorWorker::waitForAllJobsToFinish(bool &hasError) {
+    const std::scoped_lock lock(_mutex);
+
```

Comment:
Is there a risk of dead lock here since we lock again inside `deleteFinishedAsyncJobs`?

Reply:
Actually, `std::scoped_lock` embed a deadlock avoidance algorithm so it is probably ok. I am not sure however about the order of lock/unlock.

Reply:
My bad, `_mutex` is a recursive mutex. Ok for me